esbuild-runner.mjs

```diff
@@ -0,0 +1,327 @@
+    }
+
+    while(!existsSync(resolvedPath)){
+        if(countSeparators(resolvedPath) <= 1){
+            throw new Error("Could not find package.json file");
+        }
+        relativePath = Path.join("..", relativePath);
+        resolvedPath = Path.resolve(currentDir, relativePath);
+    }
+    return resolvedPath;
+}
+
+function readKeyFromPackageJson(key,pkg) {
+    let resolvedPath;
+    if(typeof pkg == "undefined" || pkg == null){
+        resolvedPath = getOwnPackageJsonPath();
+    }else{
+        resolvedPath = resolveFileFromPackage(Path.join(pkg,"package.json"));
+        if(!resolvedPath||!existsSync(resolvedPath)){
+            throw new Error("Could not find package.json file");
+        }
+    }
+
+    const packageJson = JSON.parse(readFileSync(resolvedPath, 'utf8'));
+    const keys = key.split(".");
+    let value = packageJson;
+    for(let i = 0; i < keys.length; i++) {
+        value = value[keys[i]];
+        if(value === undefined)
+            return null;
+    }
+    return value;
+}
+
+function convertToEsbuildTarget(swcTargets) {
+    const esbuildTargets = [];
+    const targetKeys = Object.keys(swcTargets);
+    for(let i = 0; i < targetKeys.length; i++) {
+        if(targetKeys[i] == "android")
+            continue;
+        esbuildTargets.push(`${targetKeys[i]}${swcTargets[targetKeys[i]]}`);
+    }
+    return esbuildTargets;
+}
+
+const parseArgs = createCommandLineArgumentsHandler({
+    "input":{
+        type: "string",
+        shortKey: true,
+        propertySyntax: true
+    },
+    "output":{
+        type: "string",
+        shortKey: true,
+        propertySyntax: true
+    },
+    "externals":{
+        type: "json",
+        shortKey: false,
+        propertySyntax: false
+    },
+    "target":{
+        type: "string",
+        shortKey: true,
+        propertySyntax: true,
+        default: "es2015"
+    },
+    "global_name": {
+        type: "string",
+        shortKey: false,
+        propertySyntax: true
+    },
+    "footer": {
+        type: "string",
+        shortKey: false,
+        propertySyntax: true
+    },
+    "watch": {
+        type: "boolean",
+        shortKey: true,
+        propertySyntax: false,
+        default: false
+    },
+    "servedir": {
+        type: "string",
+        shortKey: false,
+        propertySyntax: true
+    },
+    "minify": {
+        type: "boolean",
+        shortKey: true,
+        propertySyntax: false,
+        default: false
+    },
+    "sourcemap": {
+        type: "boolean",
+        shortKey: true,
+        propertySyntax: false,
+        default: false
+    },
+    "module": {
+        type: "boolean",
+        shortKey: true,
+        propertySyntax: false,
+        default: false
+    },
+    "debug": {
+        type: "boolean",
+        shortKey: false,
+        propertySyntax: false,
+        default: false
+    }
+});
+const params = parseArgs();
+
+async function build(params) {
+    const swcConfig = {
+        "jsc": {
```

Comment:
could this be in a /.swcrc file ?

Reply:
> could this be in a /.swcrc file ?

No, the problem with doing that is the config needs to be changed for certain files, ran into that issue just now which is why I had to push an amended commit again.

Reply:
i was thinking more of something like
```js
const swcConfig={...readFromPackage(), otherDynamicArg}
```
but nevermind.